spec.bs

```diff
@@ -1060,22 +1060,28 @@ To <dfn>obtain the plaintext payload</dfn> given an [=aggregatable report=]
 <div algorithm>
 To <dfn>encrypt the payload</dfn> given a [=byte sequence=] |plaintextPayload|,
 public key |pkR| and a [=string=] |sharedInfo|, perform the following steps.
-They return a [=byte sequence=] or an error.
+They either return a [=byte sequence=] or throw an error.
 
 1. Let |info| be the result of [=UTF-8 encoding=] the [=string/concatenation=]
     of « "`aggregation_service`", |sharedInfo| ».
-1. Let (|kem_id|, |kdf_id|, |aead_id|) be (0x0020, 0x0001, 0x0003).
-
-    Note: These indicate the HPKE algorithm identifiers, specifying the KEM
-        function as DHKEM(X25519, HKDF-SHA256), the KDF function as HKDF-SHA256
-        and the AEAD function as ChaCha20Poly1305.
-1. Let |hpkeContext| be the result of setting up an [[RFC9180|HPKE]]
-    [[RFC9180#name-encryption-to-a-public-key|sender's context]] with |pkR|,
-    |info|, |kem_id|, |kdf_id| and |aead_id|.
 1. Let |aad| be \`\` (an empty [=byte sequence=]).
-1. Let <var ignore>encryptedPayload</var> be the result of
-    [[RFC9180#name-encryption-and-decryption|encrypting]] |plaintextPayload|
-    with |hpkeContext| and |aad|.
+1. Let |ciphersuite| be (0x0020, 0x0001, 0x0003).
+
+    Note: The |ciphersuite| triple is composed of HPKE
+        [[RFC9180#name-algorithm-identifiers|algorithm identifiers]], specifying
+        the [[RFC9180#name-key-encapsulation-mechanism|KEM]] as DHKEM(X25519,
+        HKDF-SHA256), the [[RFC9180#name-key-derivation-functions-kd|KDF]]
+        function as HKDF-SHA256 and the
+        [[RFC9180#name-authenticated-encryption-wi|AEAD]] function as
+        ChaCha20Poly1305.
+1. Let |encryptedPayload| be the result of calling HPKE's
+    [[RFC9180#single-shot-apis|single-shot encryption function]]: <var
+    ignore>Seal&lt;Base&gt;</var>(|pkR|, |info|, |aad|, |plaintextPayload|).
```

Comment:
nit: I think the RFC expects `<MODE>` to be replaced with `Base` (i.e. losing the angle brackets).

But also I'm wondering if it might be clearer to keep the more prose-y description from before given that the functions defined in the RFC have important parameters omitted (no idea why they do that...).

Reply:
Definitely happy with changes to make those two points you enumerated clearer. For the second point, we should maybe be explicitly saying something like:
```
1. Let |encryptedPayload| be the concatenation of |enc| and |ct|.
```
One caveat is that the infra spec doesn't actually define concatenation for byte sequences, but I think it's pretty clear anyway. Also, given we need |enc| and |ct| separately, the benefit of using the single-shot API might be lessened. (Mainly I'm just wondering if the <MODE> parameter is confusing enough to make the separate calls better.)

Also don't feel too strongly on prose vs non-prose, but maybe we can split the difference and call it in the web spec algorithm style, but referring to the exact spec pseudocode names, e.g.:
```
1. Let (|enc|, |hpkeContext|) be the result of setting up an
  [[RFC9180|HPKE]][[RFC9180#name-encryption-to-a-public-key|sender's context]] by calling
  SetupBaseS() with a public key |pkR|, application-supplied information |info| and ciphersuite
  |ciphersuite|. If this operation fails, return an error.

  Note: Although RFC9180 omits the parameter from its pseudocode, |ciphersuite| is passed
    implicitly.
```
Wdyt?

Reply:
> One caveat is that the infra spec doesn't actually define concatenation for byte sequences, but I think it's pretty clear anyway.

Yeah, agreed. Not sure it's worth pointing out in the spec because I'm sure an implementer can figure out how to concatenate byte sequences.

> Also, given we need |enc| and |ct| separately, the benefit of using the single-shot API might be lessened. (Mainly I'm just wondering if the parameter is confusing enough to make the separate calls better.)

The single-shot encryption function returns the pair (`enc`, `ct`) and hides the HPKE context object. I think it's slightly simpler, but it does make for a kind of bloated step. I don't feel too strongly.